File: maxelement.js
// https://www.hackerrank.com/challenges/maximum-element/problem

function processData(input) {
  var lines = input.split("\n");
  var maxStack = [];
  for (var i = 1; i < lines.length; i++) {
    var arrLine = lines[i].split(" ").map(Number);
    var x = arrLine[1];
    switch (arrLine[0]) {
      case 1:
        //push
        var max = maxStack[maxStack.length - 1];
        if (maxStack.length > 0) {
          max < x ? maxStack.push(x) : maxStack.push(max);
        } else {
          maxStack.push(x);
        }
        break;
      case 2:
        //pop
        maxStack.pop();
        break;
      case 3:
        //print max
        console.log(maxStack[maxStack.length - 1]);
        break;
      default:
    }
  }
}

File: maxelement.py
# https://www.hackerrank.com/challenges/maximum-element/problem

n = int(input())

stack = [0]

for i in range(n):
    l = list(map(int, input().split()))
    if l[0] == 1:
        num = l[1]
        stack.append(max(num, stack[-1]))  # set max element
    elif l[0] == 2:
        stack.pop()
    elif l[0] == 3:
        print(stack[-1])
